scripts/scil_score_tractogram.py

```diff
@@ -0,0 +1,544 @@
+            else:
+                save_tractogram(fc_sft, os.path.join(
+                    args.out_dir, '{}_{}_fc{}'.format(prefix_1,
+                                                      prefix_2, ext)))
+
+    no_conn_sft, _ = filter_grid_roi(sft, fused_masks,
+                                     'either_end', True)
+    no_conn_streamlines = no_conn_sft.streamlines
+    tmp_sft, _ = filter_grid_roi(sft, fused_masks,
+                                 'either_end', False)
+    one_conn_sft, _ = filter_grid_roi(tmp_sft, fused_masks,
+                                      'both_ends', True)
+    one_conn_streamlines = one_conn_sft.streamlines
+    no_conn_streamlines.extend(no_conn_streamlines)
+    no_conn_streamlines.extend(one_conn_streamlines)
+
+    final_results = {}
+    no_conn_sft = StatefulTractogram.from_sft(no_conn_streamlines, sft)
+    save_tractogram(no_conn_sft, os.path.join(
+        args.out_dir, 'nc{}'.format(ext)))
+
+    # Total number of streamlines for each category
+    # and statistic that are not 'bundle-wise'
+    tc_streamlines_count = len(list(itertools.chain(*tc_streamlines_list)))
+    fc_streamlines_count = len(list(itertools.chain(*fc_streamlines_list)))
+
+    if args.wrong_path_as_separate:
+        wpc_streamlines_count = len(
+            list(itertools.chain(*wpc_streamlines_list)))
+    else:
+        wpc_streamlines_count = 0
+
+    nc_streamlines_count = len(no_conn_streamlines)
+    total_count = tc_streamlines_count + fc_streamlines_count + \
+        wpc_streamlines_count + nc_streamlines_count
+
+    final_results['tractogram_filename'] = str(args.in_tractogram)
+    final_results['tractogram_overlap'] = 0.0
+    final_results['tc_streamlines'] = tc_streamlines_count
+    final_results['fc_streamlines'] = fc_streamlines_count
+    final_results['nc_streamlines'] = nc_streamlines_count
+
+    final_results['tc_bundle'] = len([x for x in tc_streamlines_list if x])
+    final_results['fc_bundle'] = len([x for x in fc_streamlines_list if x])
+
+    final_results['tc_streamlines_ratio'] = tc_streamlines_count / total_count
+    final_results['fc_streamlines_ratio'] = fc_streamlines_count / total_count
+    final_results['nc_streamlines_ratio'] = nc_streamlines_count / total_count
+
+    if args.wrong_path_as_separate:
+        final_results['wpc_streamlines'] = wpc_streamlines_count
+        final_results['wpc_streamlines_ratio'] = \
+            wpc_streamlines_count / total_count
+        final_results['wpc_bundle'] = len(
+            [x for x in wpc_streamlines_list if x])
+
+    final_results['total_streamlines'] = total_count
+    final_results["bundle_wise"] = {}
+    final_results["bundle_wise"]["true_connections"] = {}
+    final_results["bundle_wise"]["false_connections"] = {}
+    tractogram_overlap = 0.0
+
+    # Bundle-wise statistics, useful for more complex phantom
+    for i, filename in enumerate(comb_filename):
```

Comment:
Same as above; if possible, I would try to wrap up "application logic" in separate functions (and files) to promote reusability, and keep the `main()`'s role to I/O.